packages/medusa/src/api/store/products/helpers.ts

```diff
@@ -30,3 +47,101 @@ export const maybeApplyStockLocationId = async (req: MedusaRequest, ctx) => {
 
   return entities.map((entity) => entity.stock_location_id)
 }
+
+export const wrapProductsWithTaxPrices = async <T>(
+  req: RequestWithContext<T>,
+  products: HttpTypes.StoreProduct[]
+) => {
+  // If we are missing the necessary context, we can't calculate the tax, so only `calculated_amount` will be available
+  if (
+    !req.taxContext?.taxInclusivityContext ||
+    !req.taxContext?.taxLineContext
+  ) {
+    return
+  }
+
+  // If automatic taxes are not enabled, we should skip calculating any tax
+  if (!req.taxContext.taxInclusivityContext.automaticTaxes) {
+    return
+  }
+
+  const { isTaxInclusive } = req.taxContext.taxInclusivityContext
+  const taxService = req.scope.resolve(ModuleRegistrationName.TAX)
+
+  const taxRates = (await taxService.getTaxLines(
+    products.map(asTaxItem).flat(),
+    req.taxContext.taxLineContext
+  )) as unknown as ItemTaxLineDTO[]
+
+  const taxRatesAsMap = new Map<string, ItemTaxLineDTO[]>()
+  taxRates.forEach((taxRate) => {
+    if (!taxRatesAsMap.has(taxRate.line_item_id)) {
+      taxRatesAsMap.set(taxRate.line_item_id, [])
+    }
+
+    taxRatesAsMap.get(taxRate.line_item_id)?.push(taxRate)
+  })
```

Comment:
Can just be called taxRatesMap

```suggestion
  const taxRatesMap = new Map<string, ItemTaxLineDTO[]>()
  
  taxRates.forEach((taxRate) => {
    if (!taxRatesMap.has(taxRate.line_item_id)) {
      taxRatesMap.set(taxRate.line_item_id, [])
    }

    taxRatesMap.get(taxRate.line_item_id)?.push(taxRate)
  })
```